### PR TITLE
feat(action-editor): surface sensor availability and polish picker UX

### PR DESCRIPTION
**Pull request by molant**
## Summary

A "Home Test" action configured against four door sensors never fired its burglar alarm. After tracing the problem with MCP we found the sensors had been zigbee2mqtt-offline for 7+ days — the integration was working correctly, but the editor showed every binary_sensor as if it were healthy. This PR closes that UX gap and folds in three follow-ups discovered along the way.

Four commits, each independently reviewed and approved by `pre-commit-review`:

- **`feat(action-editor): surface sensor availability + state`** — per-row state pill (open / closed / detected / clear / unavailable, device-class-aware), info button that opens HA's more-info dialog, partition that sorts unavailable sensors to the bottom of each category, header summary `Door (6) 4 unavailable`, and a matching `⚠ N of M sensors unavailable` chip on action cards in the list. Unified `isUnavailableState` predicate covers `unavailable` and `unknown` consistently across the pill, partition, header, and list-chip.
- **`feat(action-editor): table-align sensor rows; drop red separator and lifetime counter chip`** — sensor rows now use CSS grid so name / area / state / info-button align across rows. Drops the punctuation comma in the red unavailable-count span (visual noise). Removes the per-action lifetime "N triggers" chip, which users routinely misread as "N sensors that could trigger this".
- **`feat(action-editor): single-select alarm picker with responsive grid layout`** — alarms become a radiogroup (the Abode timeline only meaningfully fires one alarm code per event), laid out in 2–3 responsive columns via `grid-template-columns: repeat(auto-fit, minmax(180px, 1fr))`. Lazy migration: existing storage with >1 alarm visually shows one selected; saving persists only that one. No backend schema change.
- **`fix(websocket): exclude entities hidden in the entity registry from the sensor picker`** — picker filters `hidden_by != None` so a sensor a user has deliberately hidden can't be picked into an action. `ActionTriggerCoordinator` intentionally does *not* filter hidden — existing automations referencing a now-hidden sensor keep firing. Asymmetry documented in `docs/ARCHITECTURE.md`.

No backend Python changed beyond the one hidden-filter line + its test. The picker endpoint already returned `state` per sensor; this PR is mostly the rendering layer learning to use it.

## Test plan

- [x] Frontend unit tests: 148 tests pass (`npm test`), including new coverage for the unavailable predicate, partition order, header count, info-button event dispatch, live updates, single-select radio replace-vs-toggle semantics, responsive grid, and lazy-migration of `alarm_entity_ids`.
- [x] Backend unit tests: `pytest tests/test_websocket_api.py -k ws_entities_sensors` — 9 passed (was 8; one parametrized arm added for `RegistryEntryHider.USER` and `INTEGRATION`).
- [x] `./scripts/check.sh` — ruff / mypy / pyright / pytest + frontend gate all green.
- [x] Verified end-to-end on prod HA via the official MCP server: the four `0x000d6f...` sensors that originally surfaced this bug now render with the red "unavailable" pill, sort to the bottom of the Door category, and the action card shows `⚠ 4 of 4 sensors unavailable`. The `Security Keypad Door` sensor came up `unknown` right after a restart — exactly the edge case the predicate consolidation now treats consistently.
- [ ] Visual smoke-test on prod after merging: confirm hidden binary_sensors (toggled off via HA's entity-settings dialog) no longer appear in the picker.

### CODE REVIEW

File: frontend/src/action-editor.ts [hunk 1]
```diff
@@ -752,90 +951,137 @@ export class ActionEditor extends LitElement {
...
+            .checked=${this._selectedSensors.includes(sensor.entity_id)}
+            @change=${() => this._toggleSensor(sensor.entity_id)}
+          />
+          <span class="entity-name">${sensor.name}</span>
+          <!-- Area column always rendered (even when empty) so the
+               state-pill column lines up across rows that do and don't
+               have an area assigned. Empty cells are aria-hidden so
+               screen readers skip them — the cell exists only for
+               layout, not for semantics. -->
+          <span class="entity-area" ?aria-hidden=${!sensor.area}> ${sensor.area ?? nothing} </span>
```
[the remark this answers is not in the document]
Comment (molant) [reply frontend/src/action-editor.ts]: Addressed in 2b00d4f — switched to `aria-hidden=${!sensor.area ? 'true' : nothing}` so the attribute is either explicitly `"true"` or omitted entirely (via Lit's `nothing` sentinel). Same idiom already used elsewhere in this file. The comment block above the binding now spells out the enumerated-vs-boolean distinction so the next reader doesn't repeat the mistake.

File: frontend/src/actions-tab.ts [hunk 2]
```diff
@@ -585,8 +606,16 @@ export class ActionsTab extends LitElement {
...
-              ? html`<span class="trigger-info">${action.trigger_count} triggers</span>`
+            ${unavailableCount > 0
+              ? html`
+                  <span
+                    class="stale-warning"
+                    title="These sensors won't fire this action until they come back online."
+                  >
+                    <ha-icon icon="mdi:alert" aria-hidden="true"></ha-icon>
+                    ${unavailableCount} of ${totalCount} sensors unavailable
+                  </span>
```
[the remark this answers is not in the document]
Comment (molant) [reply frontend/src/actions-tab.ts:620]: Addressed in 2b00d4f — badge now pluralizes "sensor"/"sensors" based on `totalCount`, and the title flips between "This sensor won't fire ... until it comes back" and "These sensors won't fire ... until they come back" based on `unavailableCount`. Regression test added at `frontend/src/__tests__/actions-tab.test.ts` (`uses singular noun and title when the action references only one sensor`).

File: frontend/src/action-editor.ts [hunk 3]
```diff
@@ -716,14 +901,28 @@ export class ActionEditor extends LitElement {
...
+      isUnavailableState(getEntityState(this.hass, s.entity_id, s.state));
     const renderedCategories = nonEmptyCategories
       .map((category) => {
         const sensors = sensorsByCategory[category] ?? [];
         const filtered = isFiltering
           ? sensors.filter((s) => s.name.toLowerCase().includes(query))
           : sensors;
-        return { category, sensors, filtered };
+        // Stable partition preserves the backend's intra-group order
+        // (alphabetical by friendly_name) within each half.
```
[the remark this answers is not in the document]
Comment (molant) [reply frontend/src/action-editor.ts:920]: Addressed in 2b00d4f — went the other way: sorted the backend response so the comment's claim becomes true. `websocket_entities_sensors` now does a stable case-insensitive sort by `name` within each `device_class` at the end. Gives the picker a deterministic layout across reloads, which is also what the frontend partition (live before unavailable) was implicitly relying on. New test `test_ws_entities_sensors_sorts_by_friendly_name` covers it.

File: docs/ARCHITECTURE.md [hunk 4]
```diff
@@ -154,6 +154,8 @@ Frontend-facing command registry. All commands namespaced `abode_security/*`:
 
 Mutating commands (`create`, `update`, `delete`, `toggle`, `config/set`) require admin. `test` directly invokes an alarm trigger without persisting, for form validation.
 
+**Hidden-entity asymmetry.** `entities/sensors` filters out entries with `hidden_by != None` so users can't accidentally pick a sensor they've intentionally hidden from the UI. The `ActionTriggerCoordinator` *does not* apply the same filter — it listens on `EVENT_STATE_CHANGED` regardless of `hidden_by`, so existing actions referencing a now-hidden sensor keep firing. Hiding an entity is a UI-clutter decision, not a "retire this automation" signal.
```
[the remark this answers is not in the document]
Comment (molant) [reply docs/ARCHITECTURE.md]: Addressed in 2b00d4f — `docs/ARCHITECTURE.md:157` now uses `is not None` to match the implementation and PEP 8.
